### ISSUE TITLE
nmcli: Errors on Ubuntu 16.04 due to invalid 'autoconnect' parameter

### ISSUE THREAD

**Issue by mkbajzek** [ansible-collections/community.general#459]
##### SUMMARY

Using the `nmcli` module hits errors for `nmcli`'s inability to identify the `'autoconnect'` property, since the correct syntax is `connection.autoconnect`. This was documented in `ansible/ansible` [PR #62609](https://github.com/ansible/ansible/pull/62609), which was closed without addressing this issue because it addressed other issues that were addressed in other contributions.

I've tested the changes from that PR locally (modifying `autoconnect` within `/lib/ansiblemodules/net_tools/nmcli.py` to use @110marc 's `connection.autoconnect` syntax) and confirmed that it behaves as expected when running against both CentOS 7 and Ubuntu 16.04 (where the error has been observed using unmodified Ansible 2.9.9).

I've also tested the syntax generated by the `nmcli` module at a command prompt on Ubuntu 16.04 and confirmed that it is invalid even when Ansible is taken out of the scenario entirely, but using `connection.autoconnect` fixes the issue.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

nmcli

##### ANSIBLE VERSION

```paste below
ansible 2.9.9
  config file = None
  configured module search path = ['/Users/mbajzek/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.7 (default, Mar 10 2020, 15:43:33) [Clang 11.0.0 (clang-1100.0.33.17)]
```

##### CONFIGURATION

```paste below
DEFAULT_MANAGED_STR(<redacted>/ansible.cfg) = <redacted>
DEFAULT_ROLES_PATH(<redacted>/ansible.cfg) = ['<redacted>/roles']
```

##### OS / ENVIRONMENT

Control node OS: macOS Catalina (10.15.4)
Control node Ansible version: 2.9.9
Managed node OS: Ubuntu 16.04

##### STEPS TO REPRODUCE

Use the `nmcli` module with any arbitrary set of parameters (including the maximum number of defaults) on Ubuntu 16.04.

```yaml
---
# use nmcli module with default parameters on Ubuntu 16.04

- name: 'Use the nmcli module with default parameters'
  nmcli:
    conn_name: 'conn_name'
    state: 'present'
```

##### EXPECTED RESULTS

`nmcli` module successfully manages the specified connection and exits with a success code

##### ACTUAL RESULTS

Module fails because it considers `autoconnect` to be an invalid property, but it is always included (even when omitted in the code) due to having a default value.

```paste below
"Error: invalid <setting>.<property> 'autoconnect'."
```

**Comment by ansibullbot** [comment]
Files identified in the description:
* [`plugins/modules/net_tools/nmcli.py`](https://github.com/ansible-collections/community.general/blob/master/plugins/modules/net_tools/nmcli.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibullbot** [comment]
cc @alcamie101 @nerzhul
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Akasurde** [comment]
resolved_by_pr #478

**Comment by Akasurde** [comment]
resolved_by_pr #478

**Comment by Akasurde** [comment]
@mkbajzek, Could you please check PR #478 and let us know if it works for you ? <br><br>Thanks. <br><br> needs_info

**Comment by mkbajzek** [comment]
@Akasurde PR #478 tested on both Ubuntu 16.04 and CentOS 7 and it appears to work and resolve the issue.

I'm actually hitting a different issue now because `mtu` claims to have a default value (1500) that doesn't seem to be set correctly, but that's clearly out of scope for this issue and I can raise another.

Thanks for your help!

**Comment by mkbajzek** [comment]
Apologies: looks like @Hilzu reported the `mtu` default value issue on PR #477

Assuming that PR addresses the issue, I see no need to create a separate issue.

**Comment by mkbajzek** [comment]
@Akasurde not sure what the status is of #478 , but #508 appears to at least partially address this issue as well (given its reference to `connection.autoconnect`).

@cspeidel is 508 intended to supercede 478? If so, this issue should probably be mentioned in that PR.

**Comment by cspeidel** [comment]
@mkbajzek if you get https://github.com/ansible-collections/community.general/pull/478 merged, and I rebase https://github.com/ansible-collections/community.general/pull/508 on top (resolving any conflicts), would that work? 478 adds a lot that 508 doesn't. I'm open to any approach!

**Comment by mkbajzek** [comment]
#478 is waiting on a reply from @Akasurde. I think rebasing #508 on it after it gets merged is a reasonable approach

**Comment by bajzekm** [comment]
@Akasurde any chance of an update on this? Is there anything I can do to help get this fix through?
